### PR TITLE
Type hints for RegistrationStore

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/storage/databases/main/registration.py [hunk 1]
```diff
@@ -16,29 +16,33 @@
...
+
 THIRTY_MINUTES_IN_MS = 30 * 60 * 1000
 
 logger = logging.getLogger(__name__)
 
 
-class RegistrationWorkerStore(SQLBaseStore):
-    def __init__(self, database: DatabasePool, db_conn, hs):
+class RegistrationWorkerStore(CacheInvalidationWorkerStore):
+    def __init__(self, database: DatabasePool, db_conn, hs: "HomeServer"):
```
Comment (clokep) [inline synapse/storage/databases/main/registration.py:42]: Should we add a type for `db_conn` to each of these `__init__` methods while we're at it?

File: synapse/storage/databases/main/registration.py [hunk 2]
```diff
@@ -1016,13 +1024,56 @@ def _bg_user_threepids_grandfather_txn(txn):
 
         return 1
 
+    async def set_user_deactivated_status(
```
Comment (clokep) [inline synapse/storage/databases/main/registration.py:1027]: Why were these moved?

Comment (erikjohnston) [reply synapse/storage/databases/main/registration.py:1027]: They're used elsewhere in that class

File: synapse/storage/databases/main/registration.py (hunk 1 above)
Comment (erikjohnston) [reply synapse/storage/databases/main/registration.py]: We don't elsewhere, but that is a terrible reason not to do it.

**Review by clokep — APPROVED** [review]
